sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -620,7 +620,7 @@ private static void initializeRegistry() {
     if (!COMBINED_CACHE.containsKey(combinedPipelineOptionsInterfaces)) {
       @SuppressWarnings("unchecked")
       Class<T> allProxyClass =
-          (Class<T>) Proxy.getProxyClass(PipelineOptionsFactory.class.getClassLoader(),
```

Comment:
Please add `ReflectHelpers.findClassLoader` for `Class.forName` inside:
* `findCallersClassName`
* `printHelpUsageAndExitIfNeeded`

I would appreciate if the tests were added with this PR.